WordPressAuthenticator/NUX/NUXButton.swift

```diff
@@ -1,54 +1,149 @@
 import UIKit
 import WordPressShared
+import WordPressUI
 
 /// A stylized button used by Login controllers. It also can display a `UIActivityIndicatorView`.
-@objc open class NUXButton: NUXSubmitButton {
-    // MARK: - Configuration
-    fileprivate let horizontalInset: CGFloat = 20
-    fileprivate let verticalInset: CGFloat = 12
-    fileprivate let maxFontSize: CGFloat = 22
+@objc open class NUXButton: UIButton {
+    @objc var isAnimating: Bool {
+        return activityIndicator.isAnimating
+    }
+
+    @objc let activityIndicator: UIActivityIndicatorView = {
+        let indicator = UIActivityIndicatorView(activityIndicatorStyle: .white)
+        indicator.hidesWhenStopped = true
+        return indicator
+    }()
+
+    override open func layoutSubviews() {
+        super.layoutSubviews()
+
+        if activityIndicator.isAnimating {
+            titleLabel?.frame = CGRect.zero
+
+            var frm = activityIndicator.frame
+            frm.origin.x = (frame.width - frm.width) / 2.0
+            frm.origin.y = (frame.height - frm.height) / 2.0
+            activityIndicator.frame = frm.integral
+        }
+    }
 
-    /// Configure the appearance of the button.
+    // MARK: - Instance Methods
+
+
+    /// Toggles the visibility of the activity indicator.  When visible the button
+    /// title is hidden.
+    ///
+    /// - Parameter show: True to show the spinner. False hides it.
     ///
-    override open func configureButton() {
-        contentEdgeInsets = UIEdgeInsets(top: verticalInset, left: horizontalInset, bottom: verticalInset, right: horizontalInset)
+    open func showActivityIndicator(_ show: Bool) {
+        if show {
+            activityIndicator.startAnimating()
+        } else {
+            activityIndicator.stopAnimating()
+        }
+        configureBackgrounds()
+        configureTitleColors()
+        activityIndicator.activityIndicatorViewStyle = .gray
```

Comment:
You probably wanna move these three lines (configureBackgrounds, configureTitleColors and configureActivityIndicator) elsewhere. Perhaps  to  `awakeFromNib`.

Reason: Each time you call this, *configureBackgrounds*  will re-add the activityIndicator as a subview. This should probably print a warning or so, but it's something we should run just once.

Reply:
✔️